jasc/dialect/BUILD

```diff
@@ -0,0 +1,194 @@
+    deps = [":td_files"],
+)
+
+gentbl_cc_library(
+    name = "ops_inc_gen",
+    tbl_outs = [
+        (
+            ["-gen-op-decls"],
+            "ops.h.inc",
+        ),
+        (
+            ["-gen-op-defs"],
+            "ops.cc.inc",
+        ),
+    ],
+    tblgen = "@llvm-project//mlir:mlir-tblgen",
+    td_file = "ops.td",
+    deps = [
+        ":dialect_inc_gen",
+        ":td_files",
+    ],
+)
+
+cc_library(
+    name = "jasc_dialect_shared_library_deps",
+    deps = [
+        "@llvm-project//llvm:Support",
+        "@llvm-project//mlir:GPUDialect",
+        "@llvm-project//mlir:IR",
+        "@llvm-project//mlir:MemRefDialect",
+        "@llvm-project//mlir:TransformDialect",
+    ],
+)
+
+cc_headers_only(
+    name = "jasc_dialect_shared_library_deps_headers",
+    src = "jasc_dialect_shared_library_deps",
+)
+
+cc_headers_only(
+    name = "jasc_dialect_headers",
+    src = "dialect",
+)
+
+cc_library(
+    name = "dialect",
+    srcs = [
+        "dialect.cc",
+        "ops.cc",
+    ],
+    hdrs = [
+        "dialect.h",
+        "ops.h",
+    ],
+    deps = [
+        ":dialect_inc_gen",
+        ":ops_inc_gen",
+        ":jasc_dialect_shared_library_deps_headers",
+    ],
+    alwayslink = True,
+)
+
+cc_library(
+    name = "capi",
+    srcs = [
+        "capi.cc",
+    ],
+    hdrs = [
+        "capi.h",
+    ],
+    deps = [
+        ":dialect",
+        "@llvm-project//mlir:CAPIIRHeaders",
+    ],
+    alwayslink = True,
+)
+
+cc_library(
+    name = "capi_headers",
+    hdrs = [
+        "capi.h",
+    ],
+    deps = [
+        "@llvm-project//mlir:CAPIIRHeaders",
+    ],
+)
+
+gentbl_filegroup(
+    name = "ops_py_gen",
+    tbl_outs = [
+        (
+            [
+                "-gen-python-op-bindings",
+                "-bind-dialect=jasc",
+            ],
+            "_ops_gen.py",
+        ),
+    ],
+    tblgen = "@llvm-project//mlir:mlir-tblgen",
+    td_file = "ops_py.td",
+    deps = [
+        ":td_files",
+    ],
+)
+
+cc_library(
+    name = "jasc_dialect_shared_library",
+    srcs = [
+        ":libjascdialect.so",
+        "dialect.h",
+        "ops.h",
+    ],
+    deps = [
+        ":dialect_inc_gen",
+        ":ops_inc_gen",
+        ":jasc_dialect_shared_library_deps_headers",
+    ],
+)
+
+cc_binary(
+    name = "libjascdialect.so",
+    linkopts = [
+        "-Wl,-soname=libjascdialect.so",
+        "-Wl,-rpath='$$ORIGIN'",
+    ],
+    linkshared = 1,
+    deps = [":dialect"],
+)
+
+pybind_extension(
+    name = "bindings",
+    srcs = ["bindings.cc"],
+    deps = [
+        "@jax1//jaxlib/mlir/_mlir_libs:jaxlib_mlir_capi_shared_library",
+        ":jasc_dialect_headers",
+        "@llvm-project//mlir:CAPIIRHeaders",
+        "@llvm-project//mlir:MLIRBindingsPythonHeaders",
+        "//:mlir_lowering_shared_library",
+        "//transform_ops:jasc_transform_ops_shared_library",
+    ],
+)
+
+py_library(
+    name = "python",
+    srcs = [
+        "jasc.py",
+        # "_ods_common.py",
```

Comment:
Why is this not needed?

Reply:
In the (even more) hacky version that you reviewed, the whole target wasn't needed. I figured out a way to define the bindings in a way that is more similar to the original version, where this file is required again.